src/engine/sparqlExpressions/AggregateExpression.h

```diff
@@ -43,39 +41,60 @@ inline auto getUniqueElements = []<typename OperandGenerator>(
     }
   }
 };
-template <typename AggregateOperation, typename FinalOperation = decltype(noop)>
+
+// Class for a SPARQL expression that aggregates a given set of values to a
+// single value using `AggregateOperation`, and then applies `FinalOperation`.
+//
+// NOTE: The `FinalOperation` is typically the `identity` from above. One
+// exception is the `AvgExpression`, where the `FinalOperation` divides the
+// aggregated value (sum) by the number of elements.
+template <typename AggregateOperation,
+          typename FinalOperation = decltype(identity)>
 class AggregateExpression : public SparqlExpression {
  public:
-  // __________________________________________________________________________
+  // Create an aggregate expression, where `child` is the expression to be
+  // aggregated.
+  //
+  // For example, for `SUM(?x + 5)`, `child` is the expression for `?x + 5`,
+  // `distinct` is `false`, and `aggregateOp` is the operation for computing
+  // the sum.
```

Comment:
If you are iterating over the comments:
Typically the  `AggregateOperation` has no state and can thus be default-constructed (hence the default argument).
The only exception in the standard is `GROUP_CONCAT` , where the separator has to be passed in.